sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -146,9 +147,159 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
+         It is not recommended to modify temperature and top_p for the same completions request as the
+         interaction of these two settings is difficult to predict.
+         Supported range is [0, 1].
+         Default value is None.
+        :paramtype top_p: float
+        :keyword max_tokens: The maximum number of tokens to generate. Default value is None.
+        :paramtype max_tokens: int
+        :keyword response_format: An object specifying the format that the model must output. Used to
+         enable JSON mode. Known values are: "text" and "json_object". Default value is None.
+        :paramtype response_format: str or ~azure.ai.inference.models.ChatCompletionsResponseFormat
+        :keyword stop: A collection of textual sequences that will end completions generation. Default
+         value is None.
+        :paramtype stop: list[str]
+        :keyword stream: A value indicating whether chat completions should be streamed for this request.
+         Default value is False. If streaming is enabled, the response will be a StreamingChatCompletions.
+         Otherwise the response will be a ChatCompletions.
+        :paramtype stream: bool
+        :keyword tools: The available tool definitions that the chat completions request can use,
+         including caller-defined functions. Default value is None.
+        :paramtype tools: list[~azure.ai.inference.models.ChatCompletionsToolDefinition]
+        :keyword tool_choice: If specified, the model will configure which of the provided tools it can
+         use for the chat completions response. Is either a Union[str,
+         "_models.ChatCompletionsToolSelectionPreset"] type or a ChatCompletionsNamedToolSelection type.
+         Default value is None.
+        :paramtype tool_choice: str or ~azure.ai.inference.models.ChatCompletionsToolSelectionPreset or
+         ~azure.ai.inference.models.ChatCompletionsNamedToolSelection
+        :keyword seed: If specified, the system will make a best effort to sample deterministically
+         such that repeated requests with the
+         same seed and parameters should return the same result. Determinism is not guaranteed.
+         Default value is None.
+        :paramtype seed: int
+        :keyword model: ID of the specific AI model to use, if more than one model is available on the
+         endpoint. Default value is None.
+        :paramtype model: str
+        :return: A new ChatCompletionsClient object with defaults set.
+        :rtype: ~azure.ai.inference.ChatCompletionsClient
+        """
+        return self.__class__(
+            endpoint=self._endpoint,
+            credential=self._credential,
+            model_info=self._model_info,
+            model_extras=model_extras or self._model_extras,
+            frequency_penalty=frequency_penalty or self._frequency_penalty,
+            presence_penalty=presence_penalty or self._presence_penalty,
+            temperature=temperature or self._temperature,
```

Comment:
You need to be able to override a default with a `False`y value (e.g. changing the default temperature from 0.5 to 0).

Reply:
Good catch @johanste, thank you! Fixed and tested in this commit: https://github.com/Azure/azure-sdk-for-python/pull/36501/commits/f5f792d6ad33f474f900208b0b2d762abf783144